model-optimizer/mo/front/extractor.py

```diff
@@ -805,10 +805,16 @@ def add_input_op_input_port_without_data(graph: Graph, node_id: str, input_op, e
 
 
 def add_input_op_input_port_with_data(graph: Graph, node_id: str, input_op, edge_attrs: dict):
-    input_data_node = input_op.create_node_with_data()
-    input_node = input_data_node.in_node()
-    graph.add_edge(input_data_node.id, node_id, **edge_attrs)
-    update_ie_fields(graph.node[input_node.id])
+    assert graph.stage == 'middle', 'add_input_op_input_port_with_data() function can be used only for graph after ' \
```

Comment:
@jane-intel , could you take a look at these changes? Do we expect to keep some attributes from the data node? I see that in the modified unit test we removed `assert` checking existence of data nodes. I guess that these asserts exist for some reason.

Reply:
I checked - updated function saves necessary attributes. Related checks was added in test.